### ISSUE TITLE
Adding webhook callback to HTTP library & openapi 3.0+3.1 emitter

### ISSUE THREAD

**Issue by allenjzhang** [microsoft/typespec#4736]
Design needed.

1. decorator indicating operation is a webhook
2. allow association with operations
3. consider interaction with `@typespec/events` on event definition & declaration for webhooks
4. consider adding stronger signature definition & enforcement (linter?)
5. Backlog: client & service codegen

**Comment by vifer** [comment]
Any update on the webhook support? We’re also in a situation where having this supported would be really helpful 😄

**Comment by Kludgy4** [comment]
Agreed. This is a highly desirable feature. Most customer facing APIs at my current company use callbacks/webhooks in some way so this is blocking full migration to Typespec from raw openapi. I'll see if I can do it with @extension for now...

**Comment by randreu28** [comment]
Any updates?
